firedrake/slate/slac/compiler.py

```diff
@@ -164,7 +165,20 @@ def generate_loopy_kernel(slate_expr, tsfc_parameters=None):
     # then attach code as a c-string to the op2kernel
     code = loopy.generate_code_v2(loopy_merged).device_code()
     code = code.replace(f'void {loopy_merged.name}', f'static void {loopy_merged.name}')
-    loopykernel = op2.Kernel(code, loopy_merged.name, ldargs=["-llapack"])
+    blasdir = get_config()["options"].get("with_blas")
+    if blasdir == "download":
+        # PETSc OpenBLAS doesn't need `-llapack`
+        ldflags = []
+        include = []
+    elif blasdir:
+        # FIX ME: OpenBLAS doesn't need `-llapack`, but others might?
+        ldflags = [f"-Wl,-rpath,{blasdir}/lib", f"-L{blasdir}/lib"]
+        include = [f"{blasdir}/include"]
+    else:
+        # This case assumes we use the system (netlib) LAPACK
+        ldflags = ["-llapack"]
+        include = []
```

Comment:
I think we can get this in a consistent manner from `petsc/default/lib/petsc/conf/petscvariables` (look for `BLASLAPACK_LIB` and `BLASLAPACK_INCLUDE`).